Riot/Modules/Common/Recents/DataSources/RecentsDataSource.m

```diff
@@ -112,69 +116,61 @@ - (void)resetSectionIndexes
 
 #pragma mark - Properties
 
-- (NSArray *)invitesCellDataArray
+- (NSArray<id<MXKRecentCellDataStoring>> *)invitesCellDataArray
 {
-    return state.invitesCellDataArray;
+    return [self mapRoomSummaries:self.recentsListService.invitedRoomListData.rooms];
 }
-- (NSArray *)favoriteCellDataArray
+- (NSArray<id<MXKRecentCellDataStoring>> *)favoriteCellDataArray
 {
-    return state.favoriteCellDataArray;
+    return [self mapRoomSummaries:self.recentsListService.favoritedRoomListData.rooms];
 }
-- (NSArray *)peopleCellDataArray
+- (NSArray<id<MXKRecentCellDataStoring>> *)peopleCellDataArray
 {
-    return state.peopleCellDataArray;
+    return [self mapRoomSummaries:self.recentsListService.peopleRoomListData.rooms];
 }
-- (NSArray *)conversationCellDataArray
+- (NSArray<id<MXKRecentCellDataStoring>> *)conversationCellDataArray
 {
-    return state.conversationCellDataArray;
+    return [self mapRoomSummaries:self.recentsListService.conversationRoomListData.rooms];
```

Comment:
I wonder if we should cache all those arrays. Recreating everything every time `self.conversationCellDataArray` is called scares me.

We will not have all of them, ie potentially thousands but it should be easy to get hundreds.

Reply:
I've refactored this.